blocks/api/validation.js

```diff
@@ -401,7 +401,7 @@ export function isEquivalentHTML( actual, expected ) {
 		}
 	}
 
-	while ( ( expectedToken = getNextNonWhitespaceToken( expectedTokens ) ) ) {
+	if ( ( expectedToken = getNextNonWhitespaceToken( expectedTokens ) ) ) {
```

Comment:
These are not equivalent for me since the tested variable is changed by the call to `getNextNonWhitespaceToken`. For me the `while` checks all remaining tokens to be non-whitespace, while the `if` will only check the next one.

Reply:
No, it doesn't. If the `while` is true for the first time, it will execute its inner block. That block ends with a `return false;`, so the loop is immediately stopped (https://github.com/WordPress/gutenberg/blob/master/blocks/api/validation.js#L408).

If the intent was to actually have a loop, then the `return false;` needs to be changed instead.

Reply:
Right! I missed the "return false". I wonder if the intention was to check all the remaining tokens for a non-whitespace token and not only the next one. cc @aduth